### PR TITLE
add 020-set-gas-target playbook

### PR DESCRIPTION
**Pull request by bitwiseguy**
**Description**

Creating new playbook for op-mainnet gas target increase from 2.5Mgas/sec to 5Mgas/sec

**Tests**

None added.

**Additional context**

Related "Security Pod" support request: [link](https://github.com/orgs/ethereum-optimism/projects/56/views/12?filterQuery=gas+target&pane=issue&itemId=83546351)

**Metadata**

### CODE REVIEW

**Comment by blmalone** [comment]
@sammcingvale this pr needs a `SignFromJson.s.sol` file like this task: https://github.com/ethereum-optimism/superchain-ops/blob/main/tasks/eth/mmz-002-key-handover/SignFromJson.s.sol

File: tasks/eth/020-set-gas-target/SignFromJson.s.sol [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    uint64 initialGasLimit;
+    uint64 expectedGasLimit;
+
+    /// @notice Sets up the contract
+    function setUp() public {
+        opSystemConfigProxy = SystemConfig(readSystemConfigProxyAddress("10"));
+        opFoundationSystemConfigOwner = opSystemConfigProxy.owner();
+
+        initialGasLimit = opSystemConfigProxy.gasLimit();
+        expectedGasLimit = 90_000_000;
```
Comment (blmalone) [inline tasks/eth/020-set-gas-target/SignFromJson.s.sol:23]: Please change this @bitwiseguy accordingly.

File: tasks/eth/020-set-gas-target/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Mainnet - Ecotone Blobs Gas Configuration
```
Comment (blmalone) [inline tasks/eth/020-set-gas-target/README.md:1]: This needs updated

File: tasks/eth/020-set-gas-target/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Mainnet - Gas Target Increase
+
+Status: [READY TO SIGN]
+
+## Objective
+
+This is the playbook for updating the onchain gas limit of the `SystemConfig`. When combined with setting the `GETH_MINER_EFFECTIVEGASLIMIT = 30_000_000` env var on the sequencers, this has the effect of keeping the gas limit at 30Mgas/block and increasing the gas target from 5Mgas/block (2.5Mgas/sec) to 10Mgas/block (5Mgas/sec).
```
Comment (blmalone) [inline tasks/eth/020-set-gas-target/README.md:7]: Does this need to be `60_000_000`?

File: tasks/eth/020-set-gas-target/.env [hunk 4]
```diff
@@ -0,0 +1,4 @@
+ETH_RPC_URL="https://ethereum.publicnode.com"
+FOUNDATION_SAFE="0x847B5c174615B1B7fDF770882256e2D3E95b9D92"
```
Comment (blmalone) [inline tasks/eth/020-set-gas-target/.env:2]: We can remove `FOUNDATION_SAFE` as it's not an environment variable that's used in this execution.

File: tasks/eth/020-set-gas-target/README.md (hunk 3 above)
Comment (blmalone) [reply tasks/eth/020-set-gas-target/README.md:7]: Ah, thanks for the context. Can resolve this comment.

**Review by blmalone — APPROVED** [review]
LGTM

File: tasks/eth/020-set-gas-target/.env [hunk 5]
```diff
@@ -0,0 +1,3 @@
+ETH_RPC_URL="https://ethereum.publicnode.com"
+OWNER_SAFE="0x847B5c174615B1B7fDF770882256e2D3E95b9D92"
+SAFE_NONCE="9"
```
Comment (mds1) [inline tasks/eth/020-set-gas-target/.env:3]: For future reference, you can leave this blank to just use the current nonce

File: tasks/eth/020-set-gas-target/SignFromJson.s.sol [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    uint64 expectedGasLimit;
+
+    /// @notice Sets up the contract
+    function setUp() public {
+        opSystemConfigProxy = SystemConfig(readSystemConfigProxyAddress("10"));
+        opFoundationSystemConfigOwner = opSystemConfigProxy.owner();
+
+        initialGasLimit = opSystemConfigProxy.gasLimit();
+        expectedGasLimit = 60_000_000;
+        require(opFoundationSystemConfigOwner != address(0), "SystemConfig owner not found");
```
Comment (mds1) [inline tasks/eth/020-set-gas-target/SignFromJson.s.sol:24]: We can make this stronger with
```suggestion
        require(opFoundationSystemConfigOwner == vm.envAddress("OWNER_SAFE"), "SystemConfig owner mismatch");
```

**Pull request merged**